docs/userguide/icu_data/tracing.md

```diff
@@ -0,0 +1,132 @@
+those resources and discard the others. This guide describes how to use
+*utrace.h* to inspect resource access in real time in ICU4C.
+
+**Note:** This feature is only available in ICU4C at this time. If you are
+interested in ICU4J, please see
+[ICU-20656](https://unicode-org.atlassian.net/browse/ICU-20656).
+
+## Quick Start
+
+First, you *must* have a copy of ICU4C configured with tracing enabled.
+
+    $ ./runConfigureICU Linux --enable-tracing
+
+The following program prints resource and data usages to standard out:
+
+```cpp
+#include "unicode/brkiter.h"
+#include "unicode/errorcode.h"
+#include "unicode/localpointer.h"
+#include "unicode/utrace.h"
+
+#include <iostream>
+
+static void U_CALLCONV traceData(
+        const void *context,
+        int32_t fnNumber,
+        int32_t level,
+        const char *fmt,
+        va_list args) {
+    char        buf[1000];
+    const char *fnName;
+
+    fnName = utrace_functionName(fnNumber);
+    utrace_vformat(buf, sizeof(buf), 0, fmt, args);
+    std::cout << fnName << " " << buf << std::endl;
+}
+
+int main() {
+    icu::ErrorCode status;
+
+    const void* context;
+    utrace_setFunctions(context, nullptr, nullptr, traceData);
+    utrace_setLevel(UTRACE_VERBOSE);
+
+    // Create a new BreakIterator
+    icu::LocalPointer<icu::BreakIterator> brkitr(
+        icu::BreakIterator::createWordInstance("zh-CN", status));
+}
+```
+
+The following output is produced from this program:
+
+    FileTracer::traceOpenResFile icudt64l-brkitr/zh_CN.res
+    FileTracer::traceOpenResFile icudt64l-brkitr/zh.res
+    FileTracer::traceOpenResFile icudt64l-brkitr/root.res
+    ResourceTracer::trace (string) icudt64l-brkitr/root.res @ /boundaries/word
+    FileTracer::traceOpenDataFile icudt64l-brkitr/word.brk
+
+What this means:
+
+1. The BreakIterator constructor opened three resource files in the locale
+   fallback chain for zh_CN.
+2. One string was read from the resource bundle: the one at the resource path
+   "/boundaries/word" in brkitr/root.res.
+3. In addition, the plain data file brkitr/word.brk was opened.
+
+Based on that information, you can make a more informed decision when writing
+resource filter rules for this simple program.
+
+## Data Tracing API
+
+The `traceData` function shown above takes five arguments. The following are
+two most important for data tracing:
+
+- `fnNumber` indicates what type of data access this is.
+- `args` contains the details on which resources were accessed.
+
+**Important:** When reading from `args`, the strings are valid only within the
+scope of your `traceData` function. You should make copies the strings if you
+intend to save them for further processing.
+
+### UTRACE_RESOURCE_PATH
+
+UTRACE_RESOURCE_PATH is used to indicate that a value inside of a resource
+bundle was read by ICU code.
+
+When `fnNumber` is `UTRACE_RESOURCE_PATH`, there are three C-style strings in
+`args`:
+
+1. Data type: `"string"`, `"binary"`, `"int-vector"`, `"int"`, or `"uint"`.
```

Comment:
It makes sense to provide the types in tracing, but please document that for the purpose of resource filtering this information is irrelevant.